nuxeo-core/nuxeo-core-redis/src/main/java/org/nuxeo/ecm/core/redis/contribs/RedisDBSClusterInvalidator.java

```diff
@@ -0,0 +1,228 @@
+                    try {
+                        RedisDBSInvalidations rInvals = new RedisDBSInvalidations(nodeId, message);
+                        if (log.isTraceEnabled()) {
+                            log.trace("Receive invalidations: " + rInvals);
+                        }
+                        DBSInvalidations invals = rInvals.getInvalidations();
+                        synchronized (RedisDBSClusterInvalidator.this) {
+                            receivedInvals.add(invals);
+                        }
+                    } catch (IllegalArgumentException e) {
+                        log.error("Fail to read message: " + message, e);
+                    }
+                }
+            }, getChannelName());
+            return null;
+        });
+    }
+
+    protected String getChannelName() {
+        return namespace + INVALIDATION_CHANNEL;
+    }
+
+    protected void registerNode() {
+        startedDateTime = getCurrentDateTime();
+        List<String> keys = Collections.singletonList(getNodeKey());
+        List<String> args = Arrays.asList(STARTED_FIELD, startedDateTime,
+                Integer.valueOf(TIMEOUT_REGISTER_SECOND).toString());
+        if (log.isDebugEnabled()) {
+            log.debug("Registering node: " + nodeId);
+        }
+
+        redisExecutor.execute(jedis -> {
+            jedis.evalsha(registerSha, keys, args);
+            if (log.isInfoEnabled()) {
+                log.info("Node registered: " + nodeId);
+            }
+            return null;
+        });
+    }
+
+    protected String getNodeKey() {
+        return namespace + CLUSTER_NODES_KEY + ":" + nodeId;
+    }
+
+    @Override
+    public void close() {
+        log.debug("Closing");
+        unsubscribeToInvalidationChannel();
+        // The Jedis pool is already closed when the repository is shutdowned
+        receivedInvals.clear();
+    }
+
+    protected void unsubscribeToInvalidationChannel() {
+        subscriberThread.interrupt();
+    }
+
+    @Override
+    public DBSInvalidations receiveInvalidations() {
+        DBSInvalidations newInvals = new DBSInvalidations();
+        DBSInvalidations ret;
+        synchronized (this) {
+            ret = receivedInvals;
+            receivedInvals = newInvals;
+        }
+        return ret;
+    }
+
+    @Override
+    public void sendInvalidations(DBSInvalidations invals) {
+        RedisDBSInvalidations rInvals = new RedisDBSInvalidations(nodeId, invals);
+        if (log.isTraceEnabled()) {
+            log.trace("Sending invalidations: " + rInvals);
+        }
+        List<String> keys = Arrays.asList(getChannelName(), getNodeKey());
+        List<String> args = Arrays.asList(rInvals.serialize(), STARTED_FIELD, startedDateTime, LAST_INVAL_FIELD,
+                getCurrentDateTime(), Integer.valueOf(TIMEOUT_REGISTER_SECOND).toString());
+
+        redisExecutor.execute(jedis -> {
+            jedis.evalsha(sendSha, keys, args);
+            log.trace("invals sent");
```

Comment:
You are right it might give no performance gain here with a static message, but the logic is in place when you will add the keys to the log message. Also this trace is may be not needed, there is a monitor option to trace jedis command.